File: src/main/java/fr/alexandreguiny/lastdayonearth/variable/Category.java
package fr.alexandreguiny.lastdayonearth.variable;

public enum Category {
    TOOLS,
    CONSUMABLE,
    ARMOR,
    MATERIALS,
    OTHER;

    static public Category of(String category) {
        return switch (category) {
            case "TOOLS" -> TOOLS;
            case "CONSUMABLE" -> CONSUMABLE;
            case "ARMOR" -> ARMOR;
            case "MATERIALS" -> MATERIALS;
            default -> OTHER;
        };
    }
}

File: src/main/java/fr/alexandreguiny/lastdayonearth/variable/Category.kt
package fr.alexandreguiny.lastdayonearth.variable

import fr.alexandreguiny.lastdayonearth.variable.Color.*

enum class Category(vararg colors: Color) {
    TOOLS(GREEN, YELLOW, RED),
    CONSUMABLE(GREEN, YELLOW, RED),
    ARMOR(GREEN, YELLOW, RED),
    MATERIALS(GREEN, YELLOW, RED),
    GARAGE(GREEN, YELLOW, RED),
    OTHER(RED);

    val colors: Array<out Color>

    init {
        this.colors = colors
    }

    companion object {
        fun of(category: String): Category {
            // TODO generic
            return when (category) {
                "TOOLS" -> TOOLS
                "CONSUMABLE" -> CONSUMABLE
                "ARMOR" -> ARMOR
                "MATERIALS" -> MATERIALS
                "GARAGE" -> GARAGE
                else -> OTHER
            }
        }
    }
}